amethyst_renderer/src/sprite/prefab.rs

```diff
@@ -0,0 +1,732 @@
+use log::warn;
+use serde::{Deserialize, Serialize};
+
+use amethyst_assets::{AssetStorage, PrefabData, ProgressCounter};
+use amethyst_core::{
+    ecs::{Entity, Read, Write, WriteStorage},
+    Transform,
+};
+use amethyst_error::Error;
+
+use crate::{Sprite, SpriteRender, SpriteSheet, SpriteSheetHandle, TextureFormat, TexturePrefab};
+
+/// Represents one sprite in `SpriteList`.
+/// Positions originate in the top-left corner (bitmap image convention).
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct SpritePosition {
+    /// Horizontal position of the sprite in the sprite sheet
+    pub x: u32,
+    /// Vertical position of the sprite in the sprite sheet
+    pub y: u32,
+    /// Width of the sprite
+    pub width: u32,
+    /// Height of the sprite
+    pub height: u32,
+    /// Number of pixels to shift the sprite to the left and down relative to the entity holding it
+    pub offsets: Option<[f32; 2]>,
+}
+
+/// `SpriteList` controls how a sprite list is generated when using `Sprites::List` in a
+/// `SpriteSheetPrefab`.
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct SpriteList {
+    /// Width of the sprite sheet
+    #[serde(rename = "spritesheet_width")]
+    pub width: u32,
+    /// Height of the sprite sheet
+    #[serde(rename = "spritesheet_height")]
+    pub height: u32,
+    /// Description of the sprites
+    pub sprites: Vec<SpritePosition>,
+}
+
+/// `SpriteGrid` controls how a sprite grid is generated when using `Sprites::Grid` in a
+/// `SpriteSheetPrefab`.
+///
+/// The number of columns in the grid must always be provided, and one of the other fields must also
+/// be provided. The grid will be layout row major, starting with the sprite in the upper left corner,
+/// and ending with the sprite in the lower right corner. For example a grid with 2 rows and 4 columns
+/// will have the order below for the sprites.
+///
+/// ```text
+/// |---|---|---|---|
+/// | 0 | 1 | 2 | 3 |
+/// |---|---|---|---|
+/// | 4 | 5 | 6 | 7 |
+/// |---|---|---|---|
+/// ```
+#[derive(Clone, Debug, Deserialize, Serialize, Default)]
+pub struct SpriteGrid {
+    /// Height of the spritesheet in pixels.
+    #[serde(rename = "spritesheet_width")]
+    pub width: u32,
+    /// Width of the spritesheet in pixels.
+    #[serde(rename = "spritesheet_height")]
+    pub height: u32,
+    /// Specifies the number of columns in the spritesheet, this value must always be given.
+    pub columns: u32,
+    /// Specifies the number of rows in the spritesheet. If this is not given it will be calculated
+    /// using either `count` (`count / columns`), or `cell_size` (`sheet_size / cell_size`).
+    pub rows: Option<u32>,
+    /// Specifies the number of sprites in the spritesheet. If this is not given it will be
+    /// calculated using `rows` (`columns * rows`).
+    pub count: Option<u32>,
```

Comment:
would call this `sprite_count`, would avoid people needing to look up docs for clarification.
(oh, this means updating the docs and examples)